ykcapi/src/lib.rs

```diff
@@ -147,20 +147,70 @@ pub struct LiveAOTVals {
     length: usize,
 }
 
-/// Reads the stackmap and saved registers from the given address (i.e. the return address of the
-/// deoptimisation call).
+/// After a guard failure, reconstructs the stack frames and registers and then jumps back to the
+/// AOT code from where to continue.
+#[cfg(target_arch = "x86_64")]
+#[naked]
+#[no_mangle]
+pub extern "C" fn __ykrt_reconstruct_frames(
+    newframesptr: *const c_void,
+) {
+    unsafe {
+        asm!(
+            // The first 8 bytes of the new frames is the size of the map, needed for copying it
+            // over. Move it into rdx.
+            "mov rdx, [rdi]",
+            // Then adjust the address to where the new stack actually starts.
+            "add rdi, 8",
```

Comment:
Not sure I understand this bit. rdi containst the size of the map, which we then add 8 to, but the second instruction is talking about an address. What am i missing?

Reply:
We first copy the size of the map into `rdx`: `mov rdx, [rdi]`.
Then we add 8 to move the pointer of `rdi` to the actual beginning of the new stack. We then use that adjusted `rdi` as an argument for `memcpy` in order to copy over the new stack frames.

Reply:
ah, i see what's gone wrong. I thought you were adding to rdx. My bad.